### PR TITLE
Upgrade glutin and winit

### PR DESCRIPTION
**Pull request by sapir**
Possibly I haven't fixed everything correctly, but at least it compiles and I can display some images with it.

### CODE REVIEW

File: src/event.rs [hunk 1]
```diff
@@ -277,9 +298,17 @@ where
                 }
             }
         }
-        state.update(ctx)?;
-        state.draw(ctx)?;
-    }
 
-    Ok(())
+        if let Err(e) = state.update(ctx) {
```
Comment (sapir) [inline src/event.rs:302]: I think I should have moved this part into one of the event handlers, maybe MainEventsCleared?

**Comment by icefoxen** [comment]
Dang!  I was thinking about doing this and was just about to conclude it wasn't worth the work, and yet here it is.  Thank you!

The downside is, well, needing updated versions of all the gfx libraries.  Vendoring all of `gfx-rs` is a bit extreme.  I wonder if they'd accept PR's and make new releases?  I'll test this out and see how it goes.

**Comment by icefoxen** [comment]
Hmm.  The code works, at least from some basic tests.  If you want to continue on this could I ask you to fix the merge conflicts and do further work against the `devel` branch please?

**Comment by icefoxen** [comment]
Ooooh, this looks like it might be a good path forward for this: https://crates.io/crates/old_school_gfx_glutin_ext

> imbris: We switched to it in veloren since we haven't finished wgpu transition yet

**Comment by icefoxen** [comment]
Holy heck it works.  Thank you!  I've pulled this into  #837 and am carrying on development there.

**Comment by sapir** [comment]
> The downside is, well, needing updated versions of all the gfx libraries. Vendoring all of gfx-rs is a bit extreme. I wonder if they'd accept PR's and make new releases? I'll test this out and see how it goes.

Well, I started to make a PR to gfx-rs and discovered that updating breaks more stuff there, like gfx-window-vulkan, similar to the breakage here, so I left it at that.

**Pull request closed without merging**